### ISSUE TITLE
[Feature request] Flag to see if the device is connected or not.

### ISSUE THREAD

**Issue by vikasg603** [httptoolkit/httptoolkit#302]
In my use case, I have around 10 ADB devices connected to my PC and I use the HTTP toolkit with all of them.
Most of the time, I connect multiple devices and there's no way I can verify if the device is already connected or not without checking from the ADB device itself.

I remember in HTTP toolkit server code, using some GraphQL query, we can check all the connected devices but HTTPToolkit desktop doesn't have any such option to check it directly.

Can we expect something similar in the HTTP Toolkit?

**Comment by pimterry** [comment]
Hmm, interesting idea!

Right now, unfortunately the data isn't actually available for this. We do have a list of devices connected to ADB (which are all shown when you click the ADB interception button whenever there's more than one device available) but we don't know which ones are currently intercepting their traffic. We do know when we activate them of course, but unfortunately we don't have any way of knowing when they're deactivated, either explicitly or because the device is restarted etc.

We'd have to track that somehow, or provide an API in the Android app that would allow the desktop app to poll its connection status periodically, or find a reliable way for the Android app to tell the desktop when it disconnects (tricky, if you want to handle reboots or airplane mode correctly).

So yes, this is interesting, but I'm afraid it's not easy!

In your case, it sounds like you're using this in an automation setup, is that right? Otherwise I imagine looking at the device to check its status wouldn't be a problem. Can you explain more about why you'd use this and how? Would an ADB-based API to let you query the app's connection status for yourself be useful?

**Comment by vikasg603** [comment]
Yes in my case, adb based connectivity check will work, since it's just a part of the automation I am doing.

**Comment by pimterry** [comment]
I've done a bit of research. I think it's difficult to provide a direct API into the app, but actually it looks like you can test this already yourself by running `ifconfig tun0` via ADB. If that succeeds & prints output, then a VPN is connected (i.e. the HTTP Toolkit app). If it fails (exit code 1, printing "ifconfig: tun0: No such device") then it's not running.

Would that work for your automation?
